libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -130,9 +130,13 @@ void CompilerUtils::accessCalldataTail(Type const& _type)
 	// returns the absolute offset of the tail in "base_ref"
 	m_context.appendInlineAssembly(Whiskers(R"({
 		let rel_offset_of_tail := calldataload(ptr_to_tail)
-		if iszero(slt(rel_offset_of_tail, sub(sub(calldatasize(), base_ref), sub(<neededLength>, 1)))) { revert(0, 0) }
+		if iszero(slt(rel_offset_of_tail, sub(sub(calldatasize(), base_ref), sub(<neededLength>, 1)))) { <revertString> }
```

Comment:
@leonardoalt Is this the one you're having trouble to trigger? It's a difficult one indeed and I still don't think it can be triggered without ABIEncoderV2 types.
You can check by commenting out the line - that makes ``SolidityEndToEndTest/calldata_array_dynamic_invalid_static_middle`` and ``semanticTests/abiEncoderV2/calldata_array_dynamic_static_short_decode`` fail (both ABIEncoderV2 cases) - so those test cases indeed do trigger this.

Reply:
These are basically the nice cases that can only happen for three-dimensional arrays that I keep warning people not to forget about :-D.

Reply:
We could actually look into whether we can actually drop the line at least for some types.